handlers/handler-router/src/main/java/org/apache/servicecomb/router/custom/RouterInvokeFilter.java

```diff
@@ -68,67 +63,34 @@ public boolean enabled() {
   @Override
   public Response afterReceiveRequest(Invocation invocation,
       HttpServletRequestEx httpServletRequestEx) {
-    if (!isHaveHeadersRule()) {
+    if (!RouterRuleCache.isServerContainRule(invocation.getMicroserviceName())) {
       return null;
     }
-    if (!RouterRuleCache.isServerContainRule(invocation.getMicroserviceName())) {
+    List<String> headerRules = loadHeaders();
+    if (headerRules == null) {
       return null;
     }
-    if (loadHeaders()) {
-      Map<String, String> headerMap = getHeaderMap(httpServletRequestEx);
-      try {
-        invocation.addContext(ROUTER_HEADER, JsonUtils.OBJ_MAPPER.writeValueAsString(headerMap));
-      } catch (JsonProcessingException e) {
-        LOGGER.error("canary context serialization failed");
-      }
+    Map<String, String> headerMap = getHeaderMap(httpServletRequestEx, headerRules);
+    try {
+      invocation.addContext(ROUTER_HEADER, JsonUtils.OBJ_MAPPER.writeValueAsString(headerMap));
+    } catch (JsonProcessingException e) {
+      LOGGER.error("router context serialization failed");
     }
     return null;
```

Comment:
This modification is not good. Try find the root cause and fixed the condition why memory leak and do not modifify the logic of dynamicy config cache.